crates/next-custom-transforms/src/transforms/server_actions.rs

```diff
@@ -310,55 +315,210 @@ impl<C: Comments> ServerActions<C> {
         })
     }
 
-    // Check if the function or arrow function is an action or cache function
-    fn get_body_info(&mut self, maybe_body: Option<&mut BlockStmt>) -> BodyInfo {
-        let mut is_action_fn = false;
-        let mut cache_kind = None;
+    fn get_directive_for_stmt(
+        &mut self,
+        stmt: &Stmt,
+        directive: &mut Option<Directive>,
```

Comment:
i don't think "out parameters" like this are idiomatic. why not just return an `Option<Directive>`?

Reply:
I tried that actually, but it's tricky without introducing a bunch of cloning, because the whole function is intended to be used in the context of iterating over statements. So we need to keep track of any previous directives to emit the error when we find another one. And we need to keep track of whether non-directive statements were found, so we can emit that error. We also can't extract the whole iteration into a shared function because in the module case we're actually iterating over module items with statements and not the statements directly.

Do you have an idea how to improve this?

Side note: passing in mutable params like these is pretty common at least in this file. The whole AST visitor pattern expects you to mutate nodes in the visitor functions, for example.